ase/src/main/java/com/ase/ase/rest/controller/DistributionCovidController.java

```diff
@@ -1,25 +1,51 @@
 package com.ase.ase.rest.controller;
 
+import com.ase.ase.dao.SexAndAgeDistributionRepository;
 import com.ase.ase.rest.response.SexDistribution;
-import com.ase.ase.rest.response.DailySexDistributionPerProvinceDto;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
+import org.springframework.beans.factory.annotation.Autowired;
 
-import java.util.Arrays;
+import java.util.*;
 
 @CrossOrigin("*")
 @RestController
 @RequestMapping("/distribution")
 public class DistributionCovidController {
 
+    @Autowired
+    private SexAndAgeDistributionRepository sexAndAgeDistributionRepository;
+  
     @CrossOrigin
-    @GetMapping(value = "/sex/{province-id}", produces = "application/json")
+    @GetMapping(value = "/sex/", produces = "application/json")
     @ResponseBody
-    public ResponseEntity<DailySexDistributionPerProvinceDto> listSexDistributionFor(@PathVariable("province-id") int provinceId) {
-        return ResponseEntity.ok(
-                new DailySexDistributionPerProvinceDto(provinceId, Arrays.asList(
-                        new SexDistribution(100, 150, 4, 2)
-                ))
-        );
+    public ResponseEntity getSexDistributionBy(@RequestParam("area") Set<Integer> areas) {
+        String g = sexAndAgeDistributionRepository.getSexDistributionBy(areas);
```

Comment:
We do not need an extra variable, just return ...ok(sesAndAge...getSexDistrBy(areas)); Do this in every method